src/python/pants/bin/goal_runner.py

```diff
@@ -101,6 +101,10 @@ def _init_graph(self, use_engine, pants_ignore_patterns, build_ignore_patterns,
                                                 exclude_target_regexps=exclude_target_regexps,
                                                 subproject_roots=subproject_build_roots)
       )
+
+      # Set panic handler on native library.
+      graph_helper.scheduler.set_panic_handler()
```

Comment:
I created native object in this place, since I can pass it to setup_legacy_graph method.